File: ROT13.cpp
#include <iostream>

using namespace std;

void cipherEncryption(){
    cout << "Message can only be alphabetic" << endl;
    string message;
    cout << "Enter message: ";
    getline(cin, message);
    cin.ignore();

    //message to upper case
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    int key = 13;
    string encrypText = "";

    for(int i = 0; i < message.length(); i++){
        int temp = message[i] + key;
        if(message[i] == 32){
            encrypText += " ";
        } else if(temp > 90){
            temp -= 26;
            encrypText += (char)temp;
        } else {
            encrypText += (char)temp;
        } //if-else
    } //for

    cout << "Encrypted Text: " << encrypText;
}

void cipherDecryption(){
    cout << "Message can only be alphabetic" << endl;
    string message;
    cout << "Enter message: ";
    getline(cin, message);
    cin.ignore();

    //message to upper case
    for(int i = 0; i < message.length(); i++){
        message[i] = toupper(message[i]);
    }

    int key = 13;
    string decrypText = "";

    for(int i = 0; i < message.length(); i++){
        int temp = message[i] - key;
        if(message[i] == 32){
            decrypText += " ";
        } else if(temp < 65){
            temp += 26;
            decrypText += (char)temp;
        } else {
            decrypText += (char)temp;
        } //if-else
    } //for

    cout << "Decrypted Text: " << decrypText;
}

int main()
{
    int choice;
    cout << "1. Encryption\n2. Decryption\nChoose(1,2): ";
    cin >> choice;
    cin.ignore();

    if(choice == 1){
        cout << endl << "---Encryption---" << endl;
        cipherEncryption();

    } else if(choice == 2){
        cout << endl << "---Decryption---" << endl;
        cipherDecryption();

    } else {
        cout << endl << "Wrong Choice" << endl;
    }

    return 0;
}

File: ROT13.py
def cipher_encryption():
    print("Message can only be alphabetic")
    message = input("Enter message: ").upper()

    key = 13
    encryp_text = ""

    for i in range(len(message)):
        temp = ord(message[i]) + key
        if ord(message[i]) == 32:
            encryp_text += " "
        elif temp > 90:
            temp -= 26
            encryp_text += chr(temp)
        else:
            encryp_text += chr(temp)
        # if-else
    # for

    print("Encrypted Text: {}".format(encryp_text))


def cipher_decryption():
    print("Message can only be alphabetic")
    message = input("Enter message: ").upper()

    key = 13
    decryp_text = ""

    for i in range(len(message)):
        temp = ord(message[i]) - key
        if ord(message[i]) == 32:
            decryp_text += " "
        elif temp < 65:
            temp += 26
            decryp_text += chr(temp)
        else:
            decryp_text += chr(temp)
            # if-else
    # for

    print("Decrypted Text: {}".format(decryp_text))


def main():
    choice = int(input("1. Encryption\n2. Decryption\nChoose(1,2): "))
    if choice == 1:
        print("---Encryption---")
        cipher_encryption()

    elif choice == 2:
        print("---Decryption---")
        cipher_decryption()

    else:
        print("Invalid Choice")


if __name__ == "__main__":
    main()
